webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchService.scala

```diff
@@ -86,10 +90,14 @@ class FullQuickSearchService(implicit
       results <- ZIO.foreach(QSBackend.values)(b => {
                    b.search(query)
                      .tap(results => logger.debug(s"  - [${b}] found ${results.size} results"))
+                     .map(limit match {
+                       case Some(value) => _.take(value)
+                       case None        => identity
+                     })
                      .chainError(s"Error with quicksearch bachend ${b}")
                  })
     } yield {
-      results.toSet.flatten.take(query.limit)
+      results.flatten.toSet
```

Comment:
This will be supposed to be more efficient, as results from different backends are unique, and flattening a seq is more efficient